src/test/scala/com/databricks/spark/sql/perf/mllib/MLLibTest.scala

```diff
@@ -0,0 +1,16 @@
+package com.databricks.spark.sql.perf.mllib
+
+import org.scalatest.FunSuite
+
+class MLLibTest extends FunSuite {
+
+  test("test MlLib benchmarks with mllib-small.yaml.") {
+    val results = MLLib.run(yamlConfig = MLLib.smallConfig)
+    val failures = results.na.drop(Seq("failure"))
+    if (failures.count > 0) {
+      val failed = failures.select("name").collect.map(_.getString(0)).mkString(", ")
+      fail(s"There were failures in the following benchmarks: $failed.")
```

Comment:
We should provide more error messages. I don't know what other columns we have in result. `failures.select(a, b, c, d).show()` would provide more info for debugging.

Reply:
This is what I get with `show`:

--------------------+--------------------+
|                name|             failure|
+--------------------+--------------------+
|com.databricks.sp...|[NullPointerExcep...|
|com.databricks.sp...|[NullPointerExcep...|
|com.databricks.sp...|[NullPointerExcep...|
|com.databricks.sp...|[NullPointerExcep...|
+--------------------+--------------------+
```

I'm not sure what the best way to display this info is. The silver lining is that when benchmarks fail, the trace is dumped to the console, so if the user knows there's been a failure, they can search through the test log to find the full trace, but it could be very far from the actual failure message.

Reply:
You can collect them to local and print each of them:

~~~scala
failures.select("name", "failure")
  .as[(String, String)]
  .collect()
  .foreach { case (name, failure) =>
    // print errors
  }
}
~~~

Another way is to use `show(truncate = false)` so it doesn't truncate the messages.

Reply:
Pushed a change to print full message for failures. This is what it looks like now:

```
There as a failure in the benchmark for com.databricks.spark.sql.perf.mllib.classification.DecisionTreeClassification:
  RuntimeException foo bar:
  com.databricks.spark.sql.perf.mllib.classification.DecisionTreeClassification$.getPipelineStage(DecisionTreeClassification.scala:13)
  com.databricks.spark.sql.perf.mllib.MLPipelineStageBenchmarkable$$anonfun$3.apply(MLPipelineStageBenchmarkable.scala:62)
  com.databricks.spark.sql.perf.mllib.MLPipelineStageBenchmarkable$$anonfun$3.apply(MLPipelineStageBenchmarkable.scala:60)
  com.databricks.spark.sql.perf.Benchmarkable$class.measureTime(Benchmarkable.scala:118)
  com.databricks.spark.sql.perf.mllib.MLPipelineStageBenchmarkable.measureTime(MLPipelineStageBenchmarkable.scala:13)
  com.databricks.spark.sql.perf.mllib.MLPipelineStageBenchmarkable.doBenchmark(MLPipelineStageBenchmarkable.scala:60)
  com.databricks.spark.sql.perf.Benchmarkable$class.benchmark(Benchmarkable.scala:50)
  com.databricks.spark.sql.perf.mllib.MLPipelineStageBenchmarkable.benchmark(MLPipelineStageBenchmarkable.scala:13)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2$$anonfun$21$$anonfun$apply$9$$anonfun$23$$anonfun$25.apply(Benchmark.scala:394)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2$$anonfun$21$$anonfun$apply$9$$anonfun$23$$anonfun$25.apply(Benchmark.scala:394)
  scala.util.Try$.apply(Try.scala:192)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2$$anonfun$21$$anonfun$apply$9$$anonfun$23.apply(Benchmark.scala:393)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2$$anonfun$21$$anonfun$apply$9$$anonfun$23.apply(Benchmark.scala:376)
  scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
  scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
  scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
  scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
  scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
  scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2$$anonfun$21$$anonfun$apply$9.apply(Benchmark.scala:376)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2$$anonfun$21$$anonfun$apply$9.apply(Benchmark.scala:368)
  scala.collection.immutable.List.map(List.scala:273)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2$$anonfun$21.apply(Benchmark.scala:368)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2$$anonfun$21.apply(Benchmark.scala:367)
  scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
  scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
  scala.collection.immutable.Range.foreach(Range.scala:160)
  scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
  scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2.apply$mcV$sp(Benchmark.scala:367)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2.apply(Benchmark.scala:329)
  com.databricks.spark.sql.perf.Benchmark$ExperimentStatus$$anonfun$2.apply(Benchmark.scala:329)
  scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
  scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
  scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
  scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
  scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
  scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
  scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
  
             
[info] - test MlLib benchmarks with mllib-small.yaml. *** FAILED ***
[info]   Unable to run all benchmarks successfully, see console output for more info. (MLLibSuite.scala:20)
```